snooty/semanticparser.py

```diff
@@ -54,3 +56,117 @@ def slug_title(self, pages: Dict[FileId, Page]) -> Dict[str, SerializableType]:
             slug_title_dict[slug] = title
 
         return {"slugToTitle": slug_title_dict}
+
+    def toctree(self, pages: Dict[FileId, Page]) -> Dict[str, SerializableType]:
+        fileid_dict = {}
+        root: Dict[str, Any] = {"toctree": []}
+
+        # Construct a {slug: fileid} mapping so that we can retrieve the full file name
+        # given a slug. We cannot use the with_suffix method since the type of the slug
+        # in find_toctree_nodes(...) is string rather than FileId.
+        for fileid in pages:
+            slug = fileid.without_known_suffix
+            fileid_dict[slug] = fileid
+
+        # Build the toctree
+        for fileid in pages:
+            child: Dict["str", Any] = {}
+            ast: Dict[str, Any] = cast(Dict[str, Any], pages[fileid].ast)
+            find_toctree_nodes(fileid, ast, pages, child, fileid_dict)
```

Comment:
nit: could look for children (In abstracted function) above find_toctree_nodes.

Reply:
I was able to remove the `for-loop` actually, so the function is much more simplified now!